src/webserv/FaucetWebApi.ts

```diff
@@ -213,137 +330,237 @@ export class FaucetWebApi {
+      ? {
+          allowed: false,
+          canBeStartedAt: time,
+        }
+      : {
+          allowed: true,
+        };
   }
 
   public async onGetSession(sessionId: string): Promise<any> {
     let session: FaucetSession;
-    if(!sessionId || !(session = ServiceManager.GetService(SessionManager).getSession(sessionId, [FaucetSessionStatus.RUNNING]))) {
+    if (
+      !sessionId ||
+      !(session = ServiceManager.GetService(SessionManager).getSession(
+        sessionId,
+        [FaucetSessionStatus.RUNNING]
+      ))
+    ) {
       return {
         status: "unknown",
-        error: "Session not found"
+        error: "Session not found",
       };
     }
 
     let sessionInfo: IClientSessionInfo;
     try {
       sessionInfo = await session.getSessionInfo();
-    } catch(ex) {
-      if(ex instanceof FaucetError) {
+    } catch (ex) {
+      if (ex instanceof FaucetError) {
         let failedCode = ex.getCode();
-        ServiceManager.GetService(FaucetProcess).emitLog(FaucetLogLevel.INFO, `[FaucetWebApi.onGetSession]: Failed getting session info: ${failedCode}; sessionId: ${sessionId}`);
+        ServiceManager.GetService(FaucetProcess).emitLog(
+          FaucetLogLevel.INFO,
+          `[FaucetWebApi.onGetSession]: Failed getting session info: ${failedCode}; sessionId: ${sessionId}`
+        );
         return {
           status: FaucetSessionStatus.FAILED,
           failedCode,
           failedReason: ex.message,
-        }
-      }
-      else {
+        };
+      } else {
         return {
           status: FaucetSessionStatus.FAILED,
           failedCode: "INTERNAL_ERROR",
           failedReason: ex.toString(),
-        }
+        };
       }
     }
 
     return sessionInfo;
   }
 
+  private async onClaimRewardByGitcoin(
+    req: IncomingMessage,
+    body: Buffer,
+    userId: string
+  ): Promise<
+    | {
+        txHash: string;
+      }
+    | FaucetHttpResponse
+  > {
+    if (req.method !== "POST")
+      return new FaucetHttpResponse(405, "Method Not Allowed");
+
+    const moduleManager = ServiceManager.GetService(ModuleManager);
+    const recurringLimitsModule =
+      moduleManager.getModule<RecurringLimitsModule>("recurring-limits");
+    const remoteIP = this.getRemoteAddr(req);
+
+    // Check limits
+    const time = await recurringLimitsModule?.getTimeToNewSessionStart(
+      userId,
+      remoteIP
+    );
+    if (time > 0) {
+      throw new Error(`User has reached the limit of sessions`);
+    }
+
+    // Check active sessions
+    const activeSessions = ServiceManager.GetService(SessionManager)
+      .getActiveSessions()
+      .filter(
+        (session) =>
+          session.getSessionStatus() === FaucetSessionStatus.RUNNING &&
+          session.getUserId() === userId
+      );
+    if (activeSessions.length > 0) {
+      throw new FaucetError(
+        "GITCOIN_CLAIM_ERROR",
+        "Please stop your active mining session before claiming Gitcoin rewards."
+      );
+    }
```

Comment:
Isn't it the same as checkIfUserCanClaimGitcoin() ?

Reply:
Thanks for noticing